ResourceStatusSystem/Command/FreeDiskSpaceCommand.py

```diff
@@ -0,0 +1,102 @@
+''' FreeDiskSpaceCommand
+
+  The Command gets the free space that is left in a DIRAC Storage Element
+
+'''
+
+from datetime                                        import datetime
+from DIRAC                                           import S_OK, S_ERROR, gConfig, gLogger
+from DIRAC.ResourceStatusSystem.Command.Command      import Command
+from DIRAC.Core.DISET.RPCClient                      import RPCClient
+from DIRAC.ResourceStatusSystem.Utilities            import CSHelpers
+from DIRAC.Resources.Storage.StorageElement          import StorageElement
+
+__RCSID__ = '$Id:  $'
+
+
+class FreeDiskSpaceCommand( Command ):
+  '''
+  Uses diskSpace method to get the free space
+  '''
+
+  def __init__( self, args = None ):
+
+    super( FreeDiskSpaceCommand, self ).__init__( args )
+
+    self.rpc = None
+    self.rsClient = None
+
+  def _prepareCommand( self ):
+    '''
+      FreeDiskSpaceCommand requires one argument:
+      - name : <str>
+    '''
+
+    if 'name' not in self.args:
+      return S_ERROR( '"name" not found in self.args' )
+    elementName = self.args[ 'name' ]
+
+    return S_OK( elementName )
+
+  def doNew( self, masterParams = None ):
+    """
+    Gets the total and the free disk space of a DIPS storage element that
+    is found in the CS and inserts the results in the SpaceTokenOccupancyCache table
+    of ResourceManagementDB database.
+    """
+
+    if masterParams is not None:
+      elementName = masterParams
+    else:
+      elementName = self._prepareCommand()
+      if not elementName[ 'OK' ]:
+        return elementName
+
+    se = StorageElement(elementName)
+
+    elementURL = se.getStorageParameters(protocol = "dips")['URLBase']
+
+    if not elementURL:
+      gLogger.info( "Not a DIPS storage element, skipping..." )
+      return S_OK()
+
+    self.rpc = RPCClient( elementURL, timeout=120 )
+    free = self.rpc.getFreeDiskSpace("/")
```

Comment:
Might be (probably is) premature optimization, but do these queries take long?
You could split `getDiskSpace` even more so that you have (private) methods that take the result of a `os.statvfs` and return the free or the total space of a disk. Then 3 public methods `getFreeDiskSpace`, `getTotalDiskSpace` and `getDiskInfo` which returns a tuple ( freeSpace, totalSpace ) (would violate SRP but I could imagine a `os.stat('/')` to run quite a while).